core/blockchain.go

```diff
@@ -666,6 +666,8 @@ func (self *BlockChain) InsertChain(chain types.Blocks) (int, error) {
 			PutTransactions(self.chainDb, block, block.Transactions())
 			// store the receipts
 			PutReceipts(self.chainDb, receipts)
+			// Write map map bloom filters
+			WriteMipmapBloom(self.chainDb, block.NumberU64(), receipts)
```

Comment:
Hmm... github only showed your first response, but not the second... Anyway, that solution would also work, but you either have to always write out the blooms, or link it to canonical changes to not lose data.